Tests/FluentAssertions.Specs/Exceptions/AsyncFunctionExceptionAssertionSpecs.cs

```diff
@@ -612,259 +581,204 @@ public async Task When_subject_throws_expected_async_exact_exception_it_should_s
+            Func<Task> action = () => asyncObject
                 .Awaiting(_ => asyncObject.ReturnTaskInt())
-                .Should().NotThrow<InvalidOperationException>();
+                .Should().NotThrowAsync<InvalidOperationException>();
 
             // Assert
-            action.Should().NotThrow();
+            await action.Should().NotThrowAsync();
         }
 
         [Fact]
-        public void When_ValueTask_async_method_of_T_succeeds_and_expected_not_to_throw_particular_exception_it_should_succeed()
+        public async Task When_ValueTask_async_method_of_T_succeeds_and_expected_not_to_throw_particular_exception_it_should_succeed()
         {
-            //-----------------------------------------------------------------------------------------------------------
             // Arrange
-            //-----------------------------------------------------------------------------------------------------------
             var asyncObject = new AsyncClass();
 
-            //-----------------------------------------------------------------------------------------------------------
             // Act
-            //-----------------------------------------------------------------------------------------------------------
-            Action action = () => asyncObject
+            Func<Task> action = () => asyncObject
                 .Awaiting(_ => asyncObject.ReturnValueTaskInt())
-                .Should().NotThrow<InvalidOperationException>();
+                .Should().NotThrowAsync<InvalidOperationException>();
 
-            //-----------------------------------------------------------------------------------------------------------
             // Assert
-            //-----------------------------------------------------------------------------------------------------------
-            action.Should().NotThrow();
+            await action.Should().NotThrowAsync();
         }
 
         [Fact]
-        public void When_async_method_of_T_throws_exception_expected_not_to_be_thrown_it_should_fail()
+        public async Task When_async_method_of_T_throws_exception_expected_not_to_be_thrown_it_should_fail()
         {
             // Arrange
             var asyncObject = new AsyncClass();
 
             // Act
-            Action action = () => asyncObject
+            Func<Task> action = () => asyncObject
                 .Awaiting(x => x.ThrowTaskIntAsync<ArgumentException>(true))
-                .Should().NotThrow<ArgumentException>();
+                .Should().NotThrowAsync<ArgumentException>();
 
             // Assert
-            action.Should().Throw<XunitException>()
+            await action.Should().ThrowAsync<XunitException>()
                 .WithMessage("Did not expect System.ArgumentException, but found System.ArgumentException*");
         }
 
         [Fact]
-        public void When_ValueTask_async_method_of_T_throws_exception_expected_not_to_be_thrown_it_should_fail()
```

Comment:
Not sure why this test was deleted?

Reply:
I do not remember. :(
I've re-added and adopted to async API.